app/organization/roles/index/route.js

```diff
@@ -3,12 +3,23 @@ import Ember from 'ember';
 export default Ember.Route.extend({
   model(){
     let organization = this.modelFor('organization');
-    return organization.get('roles');
+
+    return Ember.RSVP.hash({
+      roles: organization.get('roles'),
+      stacks: this.store.find('stack')
+    });
+  },
+
+  afterModel(model){
+    return Ember.RSVP.hash({
+      users: Ember.RSVP.all(model.roles.map(r => r.get('users'))),
+      permissions: Ember.RSVP.all(model.stacks.map(s => s.get('permissions')))
```

Comment:
since these two things are not used (just preloaded) it may be worth calling out that they are preloads in a comment.

And maybe even a `// FIXME` comment, since they are such a troll on performance.